src/Mvc/Mvc.ViewFeatures/src/ClientComponentMarker.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Components
+{
+    // **Component descriptor protocol**
+    // MVC serializes one or more components as comments in HTML.
+    // Each comment is in the form <!-- Blazor:<<Json>>--> for example { "type": "client", "typeName": "<<type>>", "assembly": "<<assembly>>", "parameterDefinitions": [{"name":"<<name>>", typeName="<<type>>", "assembly": "<<assembly>>"}], "parameterValues": [<<value>>] }
+    // Where <<Json>> has the following properties:
+    // 'type' indicates the marker type. It's value for client-side components is 'client'.
+    // 'typeName' the full type name for the rendered component.
+    // 'assembly' the assembly name for the rendered component.
+    // 'parameterDefinitions' an array that contains the definitions for the parameters including their names and types and assemblies.
```

Comment:
Why do we need this? The server knows the parameter types already.

Reply:
As per other conversation: https://github.com/aspnet/AspNetCore/pull/14465#discussion_r329495458

Reply:
When we come to making this human-writable, example:

```html
<div data-blazor-component="MyApp.MyComponent" data-parameter-MyParam1="Hello" data-parameter-MyOtherThing="{'abc': 123}"></div>
```

... then we will need either:

 * Not to allow passing parameters (which would be a bit sucky)
 * Or for the client app to be able to handle not being told the parameter type in the cases where there are matching `[Parameter]` properties on the component

However that doesn't have to be part of 3.1, as there's no server-code dependency for this.